src/gui/widgets/backupwidget.cpp

```diff
@@ -43,28 +46,66 @@ BackupWidget::~BackupWidget()
 }
 
 
-void BackupWidget::on_tableView_clicked(const QModelIndex &index) {
-    selected = model->item(index.row(), 0);
-    DEB << "Item at row:" << index.row() << "->" << selected->data(Qt::DisplayRole);
+void BackupWidget::on_tableView_clicked(const QModelIndex &index)
+{
+    selectedBackupName = model->item(index.row(), 0);
+    DEB << "Item at row:" << index.row() << "->" << selectedBackupName->data(Qt::DisplayRole);
 }
 
 void BackupWidget::on_createLocalPushButton_clicked()
```

Comment:
When creating a local backup, there is no need to use a QFileDialog. The idea is that with this button, an 'automatic' backup is created in the local backup folder (and is later displayed in the table for easy "one-click" restoration). So the logic would be something like this (similar to `ADataBaseSetup::backupOldData()`)
```cpp
void BackupWidget::on_createLocalPushButton_clicked()
{
    // create the backup name and path
    QString backup_name = QLatin1String("logbook_bak_")
            + ADateTime::toString(QDateTime::currentDateTime(), Opl::Datetime::Backup)
            + QLatin1String(".db");
    QString full_path = AStandardPaths::directory(AStandardPaths::Backup).absoluteFilePath(backup_name);
    
    // create the backup
    aDB->createBackup(full_path); // with some ifs and checks (e.g. if the filename already exists etc..
    // refresh tableview
}
```

Reply:
Ah i misunderstood the functionality.
But there should be a check for whether the current db has already been backedup?
Or at least a quick prompt to give some feedback of the action and not just silently create it.

Reply:
Yeah that would be a good idea. The question is how we would go about it, maybe we could compare the database summary and prompt the user if `summary(now) == summary(lastBackup)` or we could check if a filename already exists? In that case we would only look at timestamps in a manner of speaking, as compared to looking at database states.

Not quite sure, what do you think?

Reply:
there is the case that the current db is not backed up in a file so thats another edge case...
These kind of problems are solved with hashing. Each db is hashed and thats how you compare it. Im sure Qt offers hashing functions. So in essence
```c++
int hash_of_running_db = hashFunc(ADatabase::database());
int hash_of_backup = hashFunc(ADatabase::readDb(backup_file));
if(hash_of_running_db == hash_of_backup) {
   // dont backup
}
```
The problem when hashing is simply agreeing upon what to hash and it makes sense to be the contents

Reply:
Yeah that sounds good, we could look into [QCryptographicHash](https://doc.qt.io/qt-5/qcryptographichash.html).

But I think we should keep the option to create another backup of an equal database around. Some people don't care if it's the same, they maybe want to do let's say a weekly backup for their OCD so that they don't feel like their last backup is 'too old'. So we could do something like

```cpp
[...]
if(hash_of_running_db == hash_of_backup) {
   // QMessageBox saying "The current db is already backed up as <filename>. Are you sure you want to create another backup?"
    if (reply == QMesageBox::Yes)
        // well, okay then... let's create another file
    else
        // don't backup
}

```

Reply:
simple enough, will implement in the following commits